internal/wasm/global.go

```diff
@@ -1,125 +1,55 @@
 package wasm
 
 import (
-	"fmt"
-
 	"github.com/tetratelabs/wazero/api"
 	"github.com/tetratelabs/wazero/internal/internalapi"
 )
 
-type mutableGlobal struct {
-	internalapi.WazeroOnlyType
+// constantGlobal wraps GlobalInstance to implement api.Global.
+type constantGlobal struct {
```

Comment:
This type should have `internalapi.WazeroOnlyType` as well!

Reply:
added. neat!